autobahn/websocket/protocol.py

```diff
@@ -2577,19 +2609,33 @@ def processHandshake(self):
                 if http_headers_cnt[websocket_origin_header_key] > 1:
                     return self.failHandshake("HTTP Origin header appears more than once in opening handshake request")
                 self.websocket_origin = self.http_headers[websocket_origin_header_key].strip()
+                have_origin = True
             else:
                 # non-browser clients are allowed to omit this header
-                pass
-
-            # check allowed WebSocket origins
-            #
-            origin_is_allowed = False
-            for origin_pattern in self.allowedOriginsPatterns:
-                if origin_pattern.match(self.websocket_origin):
-                    origin_is_allowed = True
-                    break
-            if not origin_is_allowed:
-                return self.failHandshake("WebSocket connection denied: origin '{0}' not allowed".format(self.websocket_origin))
+                have_origin = False
+                # XXX so, I think RFC6454 is telling us that in this
+                # case, we should "generate a fresh globally unique
+                # identifier" as the Origin... This value is passed on
+                # as "origin" to onConnect, so conceivably
+                # implementations of that could mistake "" as the same
+                # origin as another WebSocket that also just lacks on
+                # Origin header...?
+
+            # XXX to pednatically follow RFC6454, we should probably
+            # assign websocket_origin as the triple (scheme, host,
+            # port) but that would change what's in onConnect's
+            # information (for .origin). Worth breaking the API for?
+            if have_origin:
```

Comment:
@oberstet should we do either of the above XXX things? Or should I just change the comments?